classes/query.php

```diff
@@ -1524,14 +1524,22 @@ public function process_row($row, $models, &$result)
 				continue;
 			}
 
+			// If the current model is an empty relation, still need to initialize it empty later
+			$is_relation_without_records = false;
```

Comment:
I have a bit of an issue with this double negative. Technically ok, but ihmo more complex to read than `$has_related_records = true`.